### ISSUE TITLE
Private Cluster Endpoint Output Behavior is Unexpected/Undocumented

### ISSUE THREAD

**Issue by catherinetcai** [terraform-google-modules/terraform-google-kubernetes-engine#1163]
### TL;DR

The [private cluster module's](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/tree/master/modules/beta-private-cluster) endpoint output will output the control plane public IP, even if the "enable_private_endpoint" flag is set to true. The behavior is unexpected, and there's no documentation to explain the behavior. The private endpoint is only returned when the "deploy_using_private_endpoint" is enabled.

### Expected behavior

I expected the endpoint output to be the private endpoint when I set the "enable_private_endpoint" flag to true. 

Otherwise, I expect the documentation to say that the output of endpoint will always be the public endpoint unless the "deploy_using_private_endpoint" flag is set to true.

### Observed behavior

I get the public IP for the control plane for the endpoint, even though the cluster is created with the private endpoint. Only when I set "deploy_using_private_endpoint" to true do I get back the private endpoint, which doesn't seem very obvious.

### Terraform Configuration

```hcl
module "gke" {
  source  = "terraform-google-modules/kubernetes-engine/google//modules/private-cluster"
  version = "19.0.0"

  ### CLUSTER ###
  project_id         = var.project
  name               = var.cluster_name
  network_project_id = var.shared_vpc_project_id
  release_channel    = "STABLE"

  ### NETWORK ###
  default_max_pods_per_node = var.max_pods_per_node
  deploy_using_private_endpoint = true # When this flag is set to false, I get a public endpoint output
  enable_private_endpoint       = true
  ip_range_pods                 = var.ip_range_pods_name
  ip_range_services             = var.ip_range_services_name
  master_authorized_networks    = var.master_authorized_networks
  master_ipv4_cidr_block        = var.master_cidr_block
  network                       = var.network
  region                        = var.region
  subnetwork                    = var.subnet

  ### NODE ###
  enable_private_nodes = true
  node_pools           = var.node_pools
  remove_default_node_pool = true

  grant_registry_access = true
  registry_project_ids  = [var.project]
}
```

### Terraform Version

```sh
Terraform v1.1.6
on darwin_arm64
+ provider registry.terraform.io/hashicorp/external v2.2.0
+ provider registry.terraform.io/hashicorp/google v4.11.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.8.0
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
```
```


### Additional information

My thought was that if I'm using the private cluster module, and enabling the private endpoint and the private nodepool, I expect the default behavior for the endpoint to give me back the private endpoint, not the public one, unless otherwise documented.
